azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -81,22 +83,39 @@ ServiceClientCredentials getCredentials() {
             final String msiEndpoint = getProperty(this.environment, "MSI_ENDPOINT");
             final String msiSecret = getProperty(this.environment, "MSI_SECRET");
             return new AppServiceMSICredentials(AzureEnvironment.AZURE, msiEndpoint, msiSecret);
-        } else if (this.environment.containsProperty(Constants.AZURE_CLIENTID)
+        }
+
+        final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
+                Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
+        if (this.environment.containsProperty(Constants.AZURE_CLIENTID)
                 && this.environment.containsProperty(Constants.AZURE_CLIENTKEY)) {
             LOG.debug("Will use custom credentials");
-            final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
-                    Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
             final String clientId = getProperty(this.environment, Constants.AZURE_CLIENTID);
             final String clientKey = getProperty(this.environment, Constants.AZURE_CLIENTKEY);
             return new AzureKeyVaultCredential(clientId, clientKey, timeAcquiringTimeoutInSeconds);
-        } else if (this.environment.containsProperty(Constants.AZURE_CLIENTID)) {
+        }
+
+        if (this.environment.containsProperty(Constants.AZURE_CLIENTID) &&
+                this.environment.containsProperty(Constants.AZURE_KEYVAULT_CERTIFICAT_PATH)) {
+            // Password can be empty
+            final String certPwd = this.environment.getProperty(Constants.AZURE_KEYVAULT_CERTIFICAT_PASSWORD);
+            final String certPath = getProperty(this.environment, Constants.AZURE_KEYVAULT_CERTIFICAT_PATH);
+            final Resource certResource = new DefaultResourceLoader().getResource(certPath);
+
+            LOG.debug(String.format("Read certificate from %s...", certPath));
```

Comment:
1. You can use `{}` instead of `String.format`.
2. How about make this kind of log as `info` level? This help users be aware which creadential actually used.

Reply:
Forgot to use `{}`, have increased log level to `info`.